components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -3672,6 +3833,116 @@ private void updateUserClaims(User user, Map<String, String> oldClaimList,
         carbonUM.setUserClaimValuesWithID(user.getId(), userClaimsToBeModified, null);
     }
 
+    /**
+     * Evaluate old user claims and the new claims. Then DELETE, ADD and MODIFY user claim values. The DELETE,
+     * ADD and MODIFY operations are done in the same order. Multivalued claims are treated separately.
+     *
+     * @param user                           {@link User} object.
+     * @param oldClaimList                   User claim list for the user's existing state.
+     * @param newClaimList                   User claim list for the user's new state.
+     * @param allSimpleMultiValuedClaimsList User claim list which maps to simple multi-valued attributes in SCIM schema.
+     * @throws UserStoreException Error while accessing the user store.
+     * @throws CharonException    {@link CharonException}.
+     */
+    private void updateUserClaims(User user, Map<String, String> oldClaimList,
+                                  Map<String, String> newClaimList, Map<String, String> allSimpleMultiValuedClaimsList)
+            throws UserStoreException, CharonException {
+
+        Map<String, List<String>> simpleMultiValuedClaimsToBeAdded = new HashMap<>();
+        Map<String, List<String>> simpleMultiValuedClaimsToBeRemoved = new HashMap<>();
+
+        // Prepare values to be added and removed related to simple multi-valued attributes.
+        for (String key : allSimpleMultiValuedClaimsList.keySet()) {
+            String separator = ",";
+            if (StringUtils.isNotEmpty(FrameworkUtils.getMultiAttributeSeparator())) {
+                separator = FrameworkUtils.getMultiAttributeSeparator();
+            }
+            if (oldClaimList.containsKey(key) && newClaimList.containsKey(key)) {
+                List<String> oldValue = Arrays.asList(oldClaimList.get(key).split(separator));
+                List<String> newValue = Arrays.asList(newClaimList.get(key).split(separator));
+                if (!(CollectionUtils.subtract(oldValue, newValue)).isEmpty()) {
+                    simpleMultiValuedClaimsToBeRemoved.put(key,
+                            (List<String>) CollectionUtils.subtract(oldValue, newValue));
+                }
+                if (!(CollectionUtils.subtract(newValue, oldValue)).isEmpty()) {
+                    simpleMultiValuedClaimsToBeAdded.put(key,
+                            (List<String>) CollectionUtils.subtract(newValue, oldValue));
+                }
+            } else if (oldClaimList.containsKey(key) && !newClaimList.containsKey(key)) {
+                simpleMultiValuedClaimsToBeRemoved.put(key, Arrays.asList(oldClaimList.get(key).split(separator)));
+            } else if (!oldClaimList.containsKey(key) && newClaimList.containsKey(key)) {
+                simpleMultiValuedClaimsToBeAdded.put(key, Arrays.asList(newClaimList.get(key).split(separator)));
+            }
+        }
+
+        // Prepare user claims expect multi-valued claims to be added, deleted and modified.
```

Comment:
Can we move this to the multi-line block?

Reply:
addressed in https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/266/commits/d419b5d649f8bc9093b6c4143d0fa188cce5ad99